src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java

```diff
@@ -0,0 +1,206 @@
+        assertThat(statusCode, is(equalTo(HTTP_OK)));
+
+        var actualSeries = response.getBodyObject(SerialPublicationDto.class);
+        assertThat(actualSeries, is(equalTo(expectedChannel)));
+    }
+
+    @Test
+    void shouldIncludeYearInResponse() throws IOException {
+        var year = randomYear();
+        var identifier = UUID.randomUUID().toString();
+        var input = constructRequest(year, identifier, MediaType.ANY_TYPE);
+        mockChannelFoundAndReturnExpectedResponse(year, identifier, getType());
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, SerialPublicationDto.class);
+        var actualYear = response.getBodyObject(SerialPublicationDto.class).year();
+        assertThat(actualYear, is(equalTo(year)));
+    }
+
+    @ParameterizedTest(name = "Should return requested media type \"{0}\"")
+    @MethodSource("no.sikt.nva.pubchannels.handler.TestUtils#mediaTypeProvider")
+    void shouldReturnContentNegotiatedContentWhenRequested(MediaType mediaType) throws IOException {
+        var year = randomYear();
+        var identifier = UUID.randomUUID().toString();
+
+        var input = constructRequest(year, identifier, mediaType);
+
+        final var expectedMediaType =
+            mediaType.equals(MediaType.ANY_TYPE) ? MediaType.JSON_UTF_8.toString() : mediaType.toString();
+        var expectedSeries = mockChannelFoundAndReturnExpectedResponse(year, identifier, getType());
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, SerialPublicationDto.class);
+
+        var actualSeries = response.getBodyObject(SerialPublicationDto.class);
+        assertThat(actualSeries, is(equalTo(expectedSeries)));
+        var statusCode = response.getStatusCode();
+        assertThat(statusCode, is(equalTo(HTTP_OK)));
+        var contentType = response.getHeaders().get(CONTENT_TYPE);
+        assertThat(contentType, is(equalTo(expectedMediaType)));
+    }
+
+    @Test
+    void shouldReturnChannelIdWithRequestedYearIfThirdPartyDoesNotProvideYear() throws IOException {
+        var year = randomYear();
+        var identifier = UUID.randomUUID().toString();
+
+        var input = constructRequest(year, identifier, MediaType.ANY_TYPE);
+
+        var expectedSeries = mockChannelFoundYearValueNull(year, identifier);
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, SerialPublicationDto.class);
+
+        var statusCode = response.getStatusCode();
+        assertThat(statusCode, is(equalTo(HTTP_OK)));
+
+        var actualSeries = response.getBodyObject(SerialPublicationDto.class);
+        assertThat(actualSeries, is(equalTo(expectedSeries)));
+    }
+
+    @Test
+    void shouldReturnRedirectWhenChannelRegistryReturnsRedirect() throws IOException {
```

Comment:
Same with this method, it looks like all three classes are using the same business method from `FetchByIdentifierAndYearHandler`